dts/bindings/regulator/nordic,nrf-main-supply-hv.yaml

```diff
@@ -0,0 +1,23 @@
+# Copyright (c), 2024 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Nordic nRF main supply (with normal/high-voltage supply)
+
+  The power supply consists of two regulator stages: REG0 and REG1. REG0 stage
+  is enabled when working in high-voltage mode, that is, when VDDH is used as
+  the supply pin. REG0 can operate as an LDO or, in certain SoCs, as a DC/DC
+  converter. REG1 is always enabled and can operate as an LDO or DC/DC
+  converter.
+
+compatible: "nordic,nrf-main-supply-hv"
+
+include: nordic,nrf-main-supply-common.yaml
+
+properties:
+  nordic,hv-enable:
+    type: boolean
+    description: |
+      Enable high-voltage mode, i.e. supply from VDDH pin. Note that this
+      option is not available on SoCs with QFN48 package, where VDDH is
+      internally shorted to VDD.
```

Comment:
there should be a voltage selection for this too

Reply:
See https://github.com/zephyrproject-rtos/zephyr/pull/70088/files#diff-e2b7be8df6625aa6ab076a44777f8cc8c323ec90a1c2394d7c3168a945744b5e

Reply:
this is set via UICR, so not regulator registers.

Reply:
actually I likely prefer our DT representation, let me give it a spin

Reply:
are you going to put it under the custom binding? I also think it does not fit very well with the regulator api since what you can do with it is very limited (can't turn it on an off, need to reset to change the voltage...) but it'd be cool to have it as a property here and drop the board code

Reply:
```
dcdcen: dcdcen@40000578 {
	compatible = "regulator-fixed";
	regulator-name = "dcdcen";
	reg = <0x40000578 0x1>;
};

dcdcen0: dcdcen0@40000580 {
	compatible = "nordic,hvdcdc-regulator";
	regulator-name = "dcdcen0";
	reg = <0x40000580 0x1>;
	regulator-min-microvolt = <1800000>;
	regulator-max-microvolt = <3300000>;
};
```

is not a bad description, however, the `regulator-min/max-microvolt` properties are wrong, as they are meant to limit the operating range at runtime, something first unknown at soc level and then, something you can't do in practice (it's a UICR setting). As far as I can see, REGOUT0 is not currenly handled either. Another problem is that users would be dealing with dcdcen nodelabels, but these are registers in the end. Regulators are called REG/REG0/REG1. What about:

```
regulators {
	REG0: regulator@40000580 {
		compatible = "nordic,nrf-hv-regulator";
		reg = <0x40000580 0x1>;
		regulator-name = "REG0";
		status = "disabled";
	};

	REG1: regulator@40000578 {
		compatible = "regulator-fixed";
		reg = <0x40000578 0x1>;
		regulator-name = "REG1";
		regulator-init-mode = <NRF_REG_LDO>;
	};
};
```

then, at board level:

```
/* to set DC/DC instead of LDO (default) */
&REG1 {
	regulator-init-mode = <NRF_REG_DCDC>;
};

/* to enable HV mode */
&REG0 {
	status = "okay";
};
```

in the future, we could also enable support for:

```
&REG0 {
    regulator-init-microvolt = <1800000>;
};
```

thoughts?

Reply:
Looks good